### PR TITLE
Modify setup script to make pre-commit hooks executable (needed on Mac). Fixes #823

### LINKED ISSUE #823: setup.cmd should make pre-commit hooks executable
At least on Mac, when setup.cmd runs and copies the pre-commit hook into ./git/hooks, it needs to be made executable - _after_ the copy. If there isn't already an executable file in the hooks directory, it gets copied as non-executable, regardless of the status of the original file in tools. 

We should also make the pre-commit in the serializer executable, in case it hasn't come in that way from the repo.

### PR DESCRIPTION
**Pull request by rjyounes**
Fixes #823.

### CODE REVIEW

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
       - name: Install onto-tool
         run: |
           # python -m pip install --upgrade pip
+          pip install --upgrade pip
```
Comment (Jamie-SA) [inline .github/workflows/validate.yaml:19]: This has a merge conflict. Probably best to remove the change in the validate.yaml file, and leave the changes that I made in the logo PR.
```suggestion
```

**Review by Jamie-SA — CHANGES_REQUESTED** [review]

File: tools/pre-commit [hunk 2]
```diff
@@ -30,4 +33,3 @@ if [ ! ${rc} -eq 0 ] ; then
 fi
 
 exit 0
-
```
Comment (Jamie-SA) [inline tools/pre-commit:33]: General good practice is that files should end with a newline.

**Review by Jamie-SA — APPROVED** [review]

**Comment by Jamie-SA** [comment]
@rjyounes is there a reason this PR is marked as a draft? I think it is ready to merge.

**Comment by rjyounes** [comment]
I had some trouble with it and put it in draft state. I'll have another look at it.

**Pull request merged**